### PR TITLE
Fix gateway crashloop

### PR DESCRIPTION
**Pull request by frzifus**
Closes https://github.com/os-observability/tempo-operator/issues/202

### CODE REVIEW

**Comment by frzifus** [comment]
hm.. any idea whats this: https://github.com/os-observability/tempo-operator/actions/runs/4140403329/jobs/7159026854#step:5:498
?

**Comment by frzifus** [comment]
seems cm and secert are missing for some reason.. https://github.com/os-observability/tempo-operator/actions/runs/4140582629/jobs/7159375453#step:5:750

**Comment by frzifus** [comment]
I dont get why this e2e test fails. Locally iam able to run `make prepare-e2e` and afterwards `kubectl apply -f tests/e2e/gateway/01-install.yaml`. It takes 82sec until every pod is up and running.
```
NAME                                        READY   STATUS    RESTARTS   AGE
tempo-foo-compactor-9cdb4ccf9-9s2s9         1/1     Running   0          40m
tempo-foo-distributor-66c599f8d8-nqz58      1/1     Running   0          40m
tempo-foo-gateway-95c86c7fc-lvmst           1/1     Running   0          40m
tempo-foo-ingester-0                        1/1     Running   0          40m
tempo-foo-querier-6c47659ccc-f5t7v          1/1     Running   0          40m
tempo-foo-query-frontend-6b6d94dccb-zjsfx   1/1     Running   0          40m
```

Running the same thing in CI leads to:
```
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:08:51 +0000 UTC	Normal	Pod tempo-foo-query-frontend-84dc68948d-8qxxp.spec.containers{query-frontend}		Started	Started container query-frontend		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:08:59 +0000 UTC	Normal	Pod tempo-foo-gateway-79995bc77b-2kmvg.spec.containers{tempo}		Pulled	Successfully pulled image "pavolloffay/observatorium-api:001" in 5.019094203s		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:08:59 +0000 UTC	Normal	Pod tempo-foo-gateway-79995bc77b-2kmvg.spec.containers{tempo}		Created	Created container tempo		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:08:59 +0000 UTC	Normal	Pod tempo-foo-gateway-79995bc77b-2kmvg.spec.containers{tempo}		Started	Started container tempo		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:10 +0000 UTC	Normal	PersistentVolumeClaim data-tempo-foo-ingester-0		ProvisioningSucceeded	Successfully provisioned volume pvc-91aa4573-c297-4351-9aeb-62c8c406e490		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:11 +0000 UTC	Normal	Pod tempo-foo-ingester-0		Scheduled	Successfully assigned kuttl-test-fresh-stork/tempo-foo-ingester-0 to kind-control-plane		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:13 +0000 UTC	Warning	Pod tempo-foo-compactor-6977cfff8f-w9nvn.spec.containers{tempo}		Unhealthy	Readiness probe failed: HTTP probe failed with statuscode: 503		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:13 +0000 UTC	Normal	Pod tempo-foo-ingester-0.spec.containers{tempo}		Pulled	Container image "docker.io/grafana/tempo:1.5.0" already present on machine		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:13 +0000 UTC	Normal	Pod tempo-foo-ingester-0.spec.containers{tempo}		Created	Created container tempo		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:13 +0000 UTC	Normal	Pod tempo-foo-ingester-0.spec.containers{tempo}		Started	Started container tempo		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:13 +0000 UTC	Warning	Pod tempo-foo-query-frontend-84dc68948d-8qxxp.spec.containers{query-frontend}		Unhealthy	Readiness probe failed: HTTP probe failed with statuscode: 503		
    logger.go:42: 09:16:13 | gateway | 2023-02-10 09:09:31 +0000 UTC	Warning	Pod tempo-foo-ingester-0.spec.containers{tempo}		Unhealthy	Readiness probe failed: HTTP probe failed with statuscode: 503		
    logger.go:42: 09:16:13 | gateway | Deleting namespace: kuttl-test-fresh-stork
```

Any idea @pavolloffay @andreasgerstmayr @iblancasa @rubenvp8510 ? :D

**Comment by iblancasa** [comment]
@frzifus 82 seconds locally can be... ~300? in GitHub Actions Workflow. Try to increase the KUTTL timeout to verify

**Comment by frzifus** [comment]
Its now 450s but still fails after 211s

**Comment by iblancasa** [comment]
Try setting `parallel` to `1`. Maybe we can see the problem in a clearer way

File: apis/tempo/v1alpha1/microservices_types.go [hunk 1]
```diff
@@ -102,11 +102,12 @@ type MicroservicesStatus struct {
 // ModeType is the authentication/authorization mode in which Tempo Gateway
 // will be configured.
 //
-// +kubebuilder:validation:Enum=dynamic
+// +kubebuilder:validation:Enum=static
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/microservices_types.go:105]: it should list all enum values

File: apis/tempo/v1alpha1/microservices_types.go [hunk 2]
```diff
@@ -475,12 +476,12 @@ type TempoGatewaySpec struct {
...
 
-	// JaegerQuerySpec defines Jaeger Query spefic options.
+	// Gateway defines Tempo gateway spefic options.
 	//
 	// +optional
 	// +kubebuilder:validation:Optional
-	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Jaeger Query Settings"
-	Gateway GatewaySpec `json:"jaegerQuery"`
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tempo Gateway Settings"
+	Gateway GatewaySpec `json:"tempoGateway"`
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/microservices_types.go:484]: it seems wrong to have `spec.template.gateway.tempoGateway`

**Review by pavolloffay — CHANGES_REQUESTED** [review]
I would like to see 

- [ ] unit tests for the gateway configs generation, I would as well suggest splitting generation code to a separate file and have and options struct per file
- [ ] unit tests for the gateway build

File: apis/tempo/v1alpha1/microservices_types.go (hunk 1 above)
Comment (frzifus) [reply apis/tempo/v1alpha1/microservices_types.go:105]: dynamic is not implemented. thats why i removed it from the list.

File: apis/tempo/v1alpha1/microservices_types.go (hunk 2 above)
Comment (frzifus) [reply apis/tempo/v1alpha1/microservices_types.go:484]: What would be your suggestion?

**Pull request closed without merging**